### ISSUE TITLE
engine.io integration

### ISSUE THREAD

**Issue by contra** [restify/node-restify#575]
I'm trying to use engine.io with restify and can't seem to get it to work. It seems like restify is messing with the requests that come in even though I have no handlers set up for them.

```
Uncaught Error: Can't set headers after they are sent.
      at ServerResponse.OutgoingMessage.setHeader (http.js:691:11)
      at ServerResponse.format (/Users/contra/Projects/wildfire/node_modules/restify/lib/response.js:98:10)
      at ServerResponse.send (/Users/contra/Projects/wildfire/node_modules/restify/lib/response.js:208:24)
      at emitRouteError (/Users/contra/Projects/wildfire/node_modules/restify/lib/server.js:150:13)
      at onRoute (/Users/contra/Projects/wildfire/node_modules/restify/lib/server.js:617:17)
      at Router.find (/Users/contra/Projects/wildfire/node_modules/restify/lib/router.js:479:5)
      at Server._route (/Users/contra/Projects/wildfire/node_modules/restify/lib/server.js:611:17)
      at routeAndRun (/Users/contra/Projects/wildfire/node_modules/restify/lib/server.js:570:14)
      at Server._handle (/Users/contra/Projects/wildfire/node_modules/restify/lib/server.js:590:9)
      at Server.onRequest (/Users/contra/Projects/wildfire/node_modules/restify/lib/server.js:254:14)
      at Server.EventEmitter.emit (events.js:98:17)
      at HTTPParser.parser.onIncoming (http.js:2108:12)
      at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:121:23)
      at Socket.socket.ondata (http.js:1966:22)
      at TCP.onread (net.js:527:27)
```

I see there is a special case for socket.io to make restify leave requests for it alone. https://github.com/mcavage/node-restify/blob/master/lib/server.js#L250

**Comment by micahr** [comment]
Looks like we need to do a better job of allowing web socket based routes, and not require hard coding of urls in order to get around the pre processing of request attributes. Will add this to the backlog.

**Comment by retrohacker** [comment]
This appears to have been resolved in #669

If I'm missing a nuance here, please call me out on it! We can re-open this issue and discuss!
